sphinx/domains/python.py

```diff
@@ -69,7 +69,8 @@
 
 ObjectEntry = NamedTuple('ObjectEntry', [('docname', str),
                                          ('node_id', str),
-                                         ('objtype', str)])
+                                         ('objtype', str),
+                                         ('canonical', bool)])
```

Comment:
This changes the internal structure of python domain. So this might be an incompatible change....

Reply:
Finally, I and @shimizukawa considered this is an incompatible change of Python domain's API. So this should be merged into 4.0 release.

Reply:
👍